sklearn/utils/estimator_checks.py

```diff
@@ -625,7 +648,8 @@ def check_dtype_object(name, estimator_orig):
         if "Unknown label type" not in str(e):
             raise
 
-    if name not in SUPPORT_STRING:
+    tags = _safe_tags(estimator)
+    if 'str' not in tags['X_types']:
```

Comment:
I have no idea why this passes. It's not supposed to pass because this is not true for the SUPPORT_STRING estimators?